src/test/kotlin/org/jmailen/gradle/kotlinter/functional/ReportersTest.kt

```diff
@@ -0,0 +1,216 @@
+                
+                    """.trimIndent()
+                writeText(buildScript)
+            }
+        }
+        projectRoot.resolve("src/main/kotlin/CustomClass.kt") {
+            writeText(kotlinClass("CustomClass"))
+        }
+    }
+
+    @Test
+    @DisabledOnOs(OS.WINDOWS, disabledReason = "Report file content differs on Windows due to different path separator")
+    fun `supports all types of reporters`() {
+        projectRoot.resolve("build.gradle") {
+            // language=groovy
+            val buildScript =
+                """
+
+                kotlinter {
+                    reporters = [
+                        'checkstyle',
+                        'html',
+                        'json',
+                        'plain',
+                        'sarif',
+                    ] 
+                }
+
+                """
+            appendText(buildScript)
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlin")?.outcome)
+            assertEquals(expectedEmptyPlain(), reportContent("/main-lint.txt"))
+            assertEquals(expectedEmptyCheckstyle(), reportContent("main-lint.xml"))
+            assertEquals(expectedEmptyHtml(), reportContent("/main-lint.html"))
+            assertEquals(expectedEmptyJson(), reportContent("/main-lint.json"))
+            assertTrue(reportContent("/main-lint.sarif.json").isNotBlank())
+        }
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.UP_TO_DATE, task(":lintKotlin")?.outcome)
+        }
+
+        projectRoot.resolve("src/main/kotlin/CustomClass.kt") {
+            writeText(kotlinClass("WrongClassName"))
+        }
+
+        buildAndFail("lintKotlin").apply {
+            assertEquals(TaskOutcome.FAILED, task(":lintKotlinMain")?.outcome)
+            assertEquals(expectedFailedPlain(), reportContent("/main-lint.txt"))
+            assertEquals(expectedFailedCheckstyle(), reportContent("/main-lint.xml"))
+            assertEquals(expectedFailedHtml(), reportContent("/main-lint.html"))
+            assertEquals(expectedFailedJson(), reportContent("/main-lint.json"))
+            assertTrue(reportContent("/main-lint.sarif.json").isNotBlank())
+        }
+    }
+
+    @Test
+    fun `uses reporters from overridden ktlint version`() {
+        projectRoot.resolve("build.gradle") {
+            // language=groovy
+            val buildScript =
+                """
+
+                kotlinter {
+                    reporters = ['sarif']
+                     ktlintVersion = "0.47.0"
+                }
+
+                """
+            appendText(buildScript)
+        }
+
+        build("lintKotlin").apply {
+            val reportContent = projectRoot.resolve("build/reports/ktlint/main-lint.sarif.json").readText()
+            assertTrue(reportContent.contains(""""version": "0.47.0""""))
+            assertTrue(reportContent.contains(""""semanticVersion": "0.47.0""""))
```

Comment:
I noticed ktlint version appears in generated report so I decided to use to verify the behavior changes when the `ktlintVersion` gets overriden